packages/oft-evm-upgradeable/contracts/oft/MintBurnOFTAdapterUpgradeable.sol

```diff
@@ -0,0 +1,94 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.20;
+
+/// -----------------------------------------------------------------------
+/// Imports
+/// -----------------------------------------------------------------------
+
+//  ==========  LayerZero imports    ==========
+
+import { IMintableBurnable } from "@layerzerolabs/oft-evm/contracts/interfaces/IMintableBurnable.sol";
+
+//  ==========  Internal imports    ==========
+
+import { OFTAdapterUpgradeable } from "./OFTAdapterUpgradeable.sol";
+
+/// -----------------------------------------------------------------------
+/// Contract
+/// -----------------------------------------------------------------------
+
+/**
+ * @title MintBurnOFTAdapterUpgradeable
+ * @dev MintBurnOFTAdapter with upgradeable logic.
+ */
+abstract contract MintBurnOFTAdapterUpgradeable is OFTAdapterUpgradeable {
+    /// @notice The contract responsible for minting and burning tokens.
+    IMintableBurnable public immutable minterBurner;
+
+    /**
+     * @dev Constructor for initializing the contract with token and endpoint addresses.
+     * @param _token The address of the token.
+     * @param _lzEndpoint The address of the LayerZero endpoint.
+     * @param _minterBurner The address of the contract responsible for minting and burning tokens.
+     */
+    constructor(
+        address _token,
+        address _lzEndpoint,
+        IMintableBurnable _minterBurner
```

Comment:
shouldn't this also be part of the initialize?

imo only the core LZ part `_lzEndpoint` should be in the constructor.

both `_token` and `_minterBurner` should be in the initialize i think

The initialize function correctly sets up this state in the proxy. If `_token` and `_minterBurner` were only in the constructor, their values would be lost upon upgrade.

Reply:
discussed offline